### ISSUE TITLE
Struggling to install

### ISSUE THREAD

**Issue by markpendlebury** [elParaguayo/qtile-extras#290]
I see you have a [PR](https://github.com/elParaguayo/qtile-extras/pull/284) to update the documentation since removing setup.py. I've tried the updated instructions (using pip) on main and on your docs-install branch and neither seem to work? 

when i run `pip install --user .` 

i get the following output: 

```
Processing /home/${USER}/Documents/repos/qtile-extras
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: UNKNOWN
  Building wheel for UNKNOWN (pyproject.toml) ... done
  Created wheel for UNKNOWN: filename=UNKNOWN-0.23.1.dev6+g798d084-py3-none-any.whl size=1972 sha256=82bcdca0ddf33ffc05c482b00d2c181e6c28f97d3d45e8157a98e265cc34584d
  Stored in directory: /home/${USER}/.cache/pip/wheels/57/de/50/aafcff5f6088323fa287bdbe74af132f5fbcddff1a0756eb5f
Successfully built UNKNOWN
Installing collected packages: UNKNOWN
  Attempting uninstall: UNKNOWN
    Found existing installation: UNKNOWN 0.23.1.dev6+gf2b35c5
    Uninstalling UNKNOWN-0.23.1.dev6+gf2b35c5:
      Successfully uninstalled UNKNOWN-0.23.1.dev6+gf2b35c5
Successfully installed UNKNOWN-0.23.1.dev6+g798d084
``` 

Which seems to have installed something UNKNOWN? 

Then importing qtile-extras in my config still doesn't work: 

`ModuleNotFoundError: No module named 'qtile_extras'`

Any ideas? Thanks

**Comment by elParaguayo** [comment]
What version of setuptools are you running? I can't recreate this issue but from a quick search I saw a similar problem solved by upgrading setuptools.

**Comment by markpendlebury** [comment]
Thanks for the quick reply - I'm not a python dev so kinda winging it here, can you tell me how to find my version of setuptools please?

**Comment by elParaguayo** [comment]
Run `pip install -v --user .` and have a look at the output at the beginning about installing dependencies. It should say which version of setuptools and setuptools_scm it's using.

**Comment by markpendlebury** [comment]
output from `pip show setuptools` 

```
Name: setuptools
Version: 59.6.0
Summary: Easily download, build, install, upgrade, and uninstall Python packages
Home-page: https://github.com/pypa/setuptools
Author: Python Packaging Authority
Author-email: distutils-sig@python.org
License: UNKNOWN
Location: /usr/lib/python3/dist-packages
Requires: 
Required-by:
```

**Comment by elParaguayo** [comment]
That looks a bit old and is weird because qtile-extras requires setuptools>=60.0.0.

Try `pip install --user setuptools --upgrade`

**Comment by markpendlebury** [comment]
hmm, so now i have: 

```
pip show setuptools
Name: setuptools
Version: 68.2.2
Summary: Easily download, build, install, upgrade, and uninstall Python packages
Home-page: https://github.com/pypa/setuptools
Author: Python Packaging Authority
Author-email: distutils-sig@python.org
License: 
Location: /home/mpendlebury/.local/lib/python3.10/site-packages
Requires: 
Required-by:
``` 

output of `pip install -v --user .` : 
[output.log](https://github.com/elParaguayo/qtile-extras/files/13196326/output.log)


Thanks for your help with this!

**Comment by elParaguayo** [comment]
There's something very weird going on there. The log says you've got version 68 but complains that version 59 is too old!

Can you do `pip list --user`.

**Comment by markpendlebury** [comment]
`Package    Version
---------- --------------------
cairocffi  1.6.1
cffi       1.16.0
dbus-next  0.2.3
pycparser  2.21
qtile      0.23.0
setuptools 68.2.2
UNKNOWN    0.23.1.dev6+gf2b35c5
xcffib     1.5.0`

**Comment by elParaguayo** [comment]
I've no idea why that's happening but you're actually going to have another problem: if you're using the release version of qtile (v0.23.0) then you'll need to stick to the release version of qtile-extras too. That won't solve this issue though...

**Comment by markpendlebury** [comment]
ah yeah i read that somewhere too - i think because i originally installed qtile via apt. no issue removing it though and building from git if that would be better?

**Comment by elParaguayo** [comment]
You may just be able to do `pip install https://github.com/elParaguayo/qtile-extras/releases/download/v0.23.0/qtile_extras-0.23.0-py3-none-any.whl`

**Comment by elParaguayo** [comment]
Sorry, you meant installing latest git version of qtile...

**Comment by markpendlebury** [comment]
> You may just be able to do `pip install https://github.com/elParaguayo/qtile-extras/releases/download/v0.23.0/qtile_extras-0.23.0-py3-none-any.whl`

This worked perfectly thanks!

**Comment by elParaguayo** [comment]
Great - although that just gets you the v0.23 release and doesn't solve the original issue!

**Comment by Petul** [comment]
Hey!
I just ran into the same problem and found this thread. I am running Ubuntu 22.04 at it seems that currently you get setuptools version 59 if you have installed pip using `apt install python3-pip`

Solved by uninstalling `python3-pip` and reinstalling pip with the `get-pip.py` script referenced to in the pip installation guide. [https://pip.pypa.io/en/stable/installation/](https://pip.pypa.io/en/stable/installation/)
